gtsam/geometry/Pose3.cpp

```diff
@@ -63,6 +63,92 @@ Matrix6 Pose3::AdjointMap() const {
   return adj;
 }
 
+/* ************************************************************************* */
+// Calculate AdjointMap applied to xi_b, with Jacobians
+Vector6 Pose3::Adjoint(const Vector6& xi_b, OptionalJacobian<6, 6> H_this,
+                       OptionalJacobian<6, 6> H_xib) const {
+  //  Ad   *   xi   =   [   R   0    .  [w
+  //                      [t]R  R ]      v]
+  // Declarations, aliases, and intermediate Jacobians easy to compute now
+  Vector6 result;
+  auto Rw = result.head<3>();
+  const Vector3 &w = xi_b.head<3>(), &v = xi_b.tail<3>();
+  Matrix3 Rw_H_R, Rv_H_R;
+  const Matrix3 &Rw_H_w = R_.matrix();
+  const Matrix3 &Rv_H_v = R_.matrix();
+  const Matrix3 pRw_H_Rw = skewSymmetric(t_);
+
+  // Calculations
+  Rw = R_.rotate(w, H_this ? &Rw_H_R : nullptr /*, Rw_H_w */);
+  const Vector3 Rv = R_.rotate(v, H_this ? &Rv_H_R : nullptr /*, Rv_H_v */);
+  const Vector3 pRw = cross(t_, Rw /*, pRw_H_t, pRw_H_Rw */);
+  result.tail<3>() = pRw + Rv;
+
+  // Jacobians
+  if (H_this) {
+    // pRw_H_thisv = pRw_H_t * R = [Rw]x * R = R * [w]x = Rw_H_R
+    //    where [ ]x denotes the skew-symmetric operator.
+    // See docs/math.pdf for more details.
+    const Matrix3 &pRw_H_thisv = Rw_H_R;
+    *H_this = (Matrix6() << Rw_H_R, /*               */ Z_3x3,  //
+               /*        */ pRw_H_Rw * Rw_H_R + Rv_H_R, pRw_H_thisv)
+                  .finished();
+  }
+  if (H_xib) {
+    *H_xib = (Matrix6() << Rw_H_w, /*      */ Z_3x3,  // note: this is Adjoint
```

Comment:
This is just AdjointMap

Reply:
Ya I figured we could save a function call and re-use intermediates this way (namely `[t]x`), hoping that the comment `note: this is Adjoint` was sufficient to clarify that this is just the same as `AdjointMap`

Shall I replace it with the marginally less efficient (?) call to AdjointMap or clarify the comment or ?